### ISSUE TITLE
Error in compiling LAMMPS with schnetpack interface

### ISSUE THREAD

**Issue by francescomambretti** [atomistic-machine-learning/schnetpack#520]
Hi there, I am following your guide https://schnetpack.readthedocs.io/en/latest/howtos/lammps.html. My system is Ubuntu 20.04, all CUDA stuff is already installed and working.
Towards the very end of the LAMMPS compilation, the following errors come out (see attachment):
‘int LAMMPS_NS::NeighRequest::half’ is protected within this context
‘int LAMMPS_NS::NeighRequest::full’ is protected within this context
‘int LAMMPS_NS::NeighRequest::ghost’ is protected within this context

I have no clue about how to solve this. I am working with last LAMMPS stable version (the one you mention in the tutorial is no longer available). Could you please provide any support?
Thanks

Francesco Mambretti

<img width="1377" alt="image" src="https://user-images.githubusercontent.com/48158511/232013313-98f3f15c-d581-4243-accc-8fa48fd82899.png">

**Comment by francescomambretti** [comment]
Update: with the version reported in the tutorial, it works. This is somehow limitating if one wishes to use a newer version.
A few points about the installation procedure:
- CUDA versions newer than 11.7 will not always work
- there is not always a proper cuDNN version to be installed, one has to look among the autocompletion options and choose the best one
- "conda create -n spk_lammps python=3.9 pytorch-cuda=11.7 pytorch mkl-include numpy -c pytorch -c nvidia" does not work, while "conda create -n spk_lammps python=3.9 cuda-toolkit=11.7 pytorch mkl-include numpy -c pytorch -c nvidia" does
- when dealing with spkdeploy, one has to change, inside ~/anaconda3/envs/spk_lammps/lib/python3.9/site-packages/schnetpack/data/datamodule.py, "GPUAccelerator" into "CUDAAccelerator" in the imports

**Comment by Stefaanhess** [comment]
Hi @francescomambretti,

thanks for you suggestions!
We are currently working on a fix for the LAMMPS integration with the current version.
I will link the branch here, as soon as it is available.

The CUDA and cuDNN versions seem to be quiet an issue with the LAMMPS interface. We only tested for CUDA=11.7. Maybe it would make sense to set it to 11.7 in the tutorial. I am not sure, what you mean regarding the cuDNN versions. If you want to make any changes regarding the installation tutorial, you could also directly change them in the LAMMPS tutorial and create a PR. We are very happy about any contributions.

I will also test the `conda create -n spk_lammps python=3.9 pytorch-cuda=11.7 pytorch mkl-include numpy -c pytorch -c nvidia`
command, but I can not do this on my local machine.

Are you already using pytorch lightning 2.0? `GPUAccelerator` might be deprecated with the newest version.

Best,
Stefaan

**Comment by Stefaanhess** [comment]
Thanks for the explanation. We will include this in the tutorial.

We also tested the fix for the new LAMMPS version and it seems to work well. You can find it in #522. Let me know if this fix also works for you!

**Comment by Stefaanhess** [comment]
Hi @francescomambretti,

we updated the LAMMPS interface, to also work with the new LAMMPS versions.
I also made some minor changes to the tutorial to point out more clearly, that different CUDA versions can cause unknown errors. We also changed the command, that did not work for you, although it somehow runs fine on our machine.
Since we link to the cuDNN website for the installation, I did not make any changes to the installation of cuDNN with the command line for now.
Finally I checked in on the problem with the lightning version. We already pushed a fix for the errors, that are caused by lightning 2.0. So if you use the current master branch of schnetpack, it should work well. We will also make a pipy release for this.
Thanks again for you comments and let me know if you have any further remarks on this.
If not, feel free to close the issue.

Best,
Stefaan
